File: src/domain/services/stack_resolver.py
"""Domain service: resolve TechStack to StackProfile.

Stateless factory that maps a TechStack value object to the appropriate
StackProfile strategy. No I/O, no side effects.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from src.domain.models.stack_profile import GenericProfile, PythonUvProfile

if TYPE_CHECKING:
    from src.domain.models.stack_profile import StackProfile
    from src.domain.models.tech_stack import TechStack


def resolve_profile(tech_stack: TechStack | None) -> StackProfile:
    """Map a TechStack to the corresponding StackProfile.

    Args:
        tech_stack: The tech stack from a discovery session, or None for
            old sessions that predate tech stack detection.

    Returns:
        PythonUvProfile if language is "python", GenericProfile otherwise.
    """
    if tech_stack is not None and tech_stack.language == "python":
        return PythonUvProfile()
    return GenericProfile()

File: internal/discovery/domain/stack_resolver.go
package domain

import (
	"strings"

	vo "github.com/alto-cli/alto/internal/shared/domain/valueobjects"
)

// ResolveProfile maps a TechStack to the corresponding StackProfile.
// Uses case-insensitive matching. Returns PythonUvProfile for "python",
// GoModProfile for "go", GenericProfile otherwise.
func ResolveProfile(techStack *vo.TechStack) vo.StackProfile {
	if techStack == nil {
		return vo.GenericProfile{}
	}

	switch strings.ToLower(techStack.Language()) {
	case "python":
		return vo.PythonUvProfile{}
	case "go":
		return vo.GoModProfile{}
	default:
		return vo.GenericProfile{}
	}
}
